pkg/template/api/types.go

```diff
@@ -16,6 +16,10 @@ type Template struct {
 	// Optional: Parameters is an array of Parameters used during the
 	// Template to Config transformation.
 	Parameters []Parameter `json:"parameters,omitempty"`
+
+	// Optional: ObjectLabels is a set of labels that are applied to every
+	// object during the Template to Config transformation
+	ObjectLabels map[string]string `json:"objectlabels,omitempty"`
```

Comment:
Yes, the versioned API can be labels (since that's JSON, and in JSON it's clear) and internally you can use ObjectLabels.

----- Original Message -----
> > @@ -16,6 +16,10 @@ type Template struct {
> >  	// Optional: Parameters is an array of Parameters used during the
> >  	// Template to Config transformation.
> >  	Parameters []Parameter `json:"parameters,omitempty"`
> > +
> > +	// Optional: ObjectLabels is a set of labels that are applied to every
> > +	// object during the Template to Config transformation
> > +	ObjectLabels map[string]string `json:"objectlabels,omitempty"`
> 
> @smarterclayton - by externally vs internally do you mean internal api vs
> versioned api ?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/982/files#r24905989